sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/dao/DaoFactory.java

```diff
@@ -133,4 +133,11 @@ public synchronized ChangeStreamDao getChangeStreamDao() {
     }
     return changeStreamDaoInstance;
   }
+
+  /** Resets the DaoFactory for testing. */
+  public static void reset() {
```

Comment:
This public method worries me a bit, because we will allow for users to mutate state of this instance. Other than that, it is not in a synchronized block, so it may break in a concurrent environment. Anyway we could inject a new instance of the DaoFactory on every test instead?